src/site/README.md

```diff
@@ -0,0 +1,309 @@
+1. Edits to the `static-pages` application code will refresh via hot reload
+1. After editing a template, issue a `yarn build:content` in a separate terminal so as not to disrupt your application server
+
+_Note: you may have noticed that the project also has a `yarn watch:content` task available. At the time of writing, this task has appeared to be unstable. For example, if you make multiple edits to a template file and save after edit before the former content builds have a chance to complete, your watch task may crash._
+
+</details>
+
+<details><summary>Edit using the preview server (preferred)</summary>
+
+This approach is recommended if you are making heavy changes to a template or are writing a template for a page-type without any published instances. The preview server will load data fresh from the CMS on each page request and processes it through the template at that time, meaning you don't need to wait for a content build to run to see your changes.
+
+1. Run `yarn build:webpack --env.entry=static-pages` to generate the JS/CSS files required by templates. If you make changes to the `static-pages` application code, you will need to run this command again.
+1. Determine the `entityId` of a page that uses the template you are editing.
+    - If you are interested in templating against a page already published, you can find its `entityId` by navigating to the page on the website (dev, staging, or prod will all work) and inspecting its DOM. At the top of the DOM should be a comment containing its `entityId`.
+    - If you are interested in templating against a page that has not been published, you can look up an `entityId` of that page-type using the [GraphQL Explorer](https://prod.cms.va.gov/graphql/explorer). See the section below for an example.
+1. Start the preview server via `yarn preview`. Your SOCKS proxy must be running.
+1. Navigated to `http://localhost:3001/preview?nodeId=${YOUR_ENTITY_ID}`
+1. After making a change to the template, stop your preview server and start it again. There is no hot reload for the preview server. Fortunately, it is lightweight enough to be quick to stop and start.
+
+### How to look look up an entityId
+The following query looks up the `entityId` for one page of the `entityBundle` (page-type) with value `landing_page`. This would be useful if you are interested in making changes to the `landing_page.drupal.liquid` template but are are unsure what pages on the website are instances of that page-type. Run this query using the [GraphQL Explorer](https://prod.cms.va.gov/graphql/explorer).
+
+```graphql
+{
+  nodeQuery(
+    limit: 1,
+    filter: {
+      conditions: [{field: "type", value: ["landing_page"]}]
+    }) {
+    entities {
+      entityId
+    }
+  }
+}
+```
+
+</details>
```

Comment:
I've noticed functions in various places that seems to be needed for certain pages. e.g.

- [compilePage](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/drupal/page.js#L272)
- [pipeDrupalPagesIntoMetalsmith](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/drupal/metalsmith-drupal.js#L48) has a big `switch` statement with special logic for various pages - looks like mostly sorting and data transformation.

Does this doc need to cover these?

Reply:
Maybe another time but right now my knowledge of that area is too limited for me to write about it